### PR TITLE
Onnx implementation

### PR DESCRIPTION
**Pull request by nprouvost**
Implementation of ONNX, implementation of PlotRuntimesMultipleParams instead of PlotRuntimesMultipleNetworks and PlotRuntimesMultipleCMSSW, refactoring parameters and new parameters for labels, add onnx examples and update existing tf examples, updated docu (wrong as of now in master after last refactoring)

### CODE REVIEW

File: mlprof/tasks/parameters.py [hunk 1]
```diff
@@ -72,38 +103,124 @@ def __init__(self, *args, **kwargs):
...
+        if self.model_names[0] == law.NO_STR:
+            if self.model_labels[0] != law.NO_STR:
+                if len(self.model_files) != len(self.model_labels):
+                    raise ValueError("the length of model_files and model_labels muss be the same")
+        elif self.model_labels[0] == law.NO_STR:
+            if len(self.model_files) != len(self.model_names):
+                raise ValueError("the length of model_files and model_names muss be the same")
+        else:
+            if len({len(self.model_files), len(self.model_names), len(self.model_labels)}) != 1:
+                raise ValueError("the length of model_names, model_files and model_labels muss be the same")
```
Comment (riga) [inline mlprof/tasks/parameters.py]: This could use some de-nesting.

File: mlprof/tasks/parameters.py [hunk 2]
```diff
@@ -12,6 +12,47 @@
 from mlprof.tasks.base import BaseTask
 
 
+class Model():
+
+    def __init__(self, model_file: str, name, label, **kwargs):
+
```
Comment (riga) [inline mlprof/tasks/parameters.py:18]: Very strictly speaking not necessary, but we should still add a `super()` here.

**Review by riga — CHANGES_REQUESTED** [review]
Very nice!

I have two minor comments (inline) and two additional suggestions.

1. There is now quite a lot of code duplication between the ONNX and TF runtime measurement plugins. Let's have the overlap organized in a common base class.
2. We should also be able to load models from yaml files instead of json.

Let's chat privately on how to divide the work.

File: mlprof/tasks/parameters.py (hunk 1 above)
Comment (nprouvost) [reply mlprof/tasks/parameters.py]: done

File: mlprof/tasks/parameters.py (hunk 2 above)
Comment (nprouvost) [reply mlprof/tasks/parameters.py:18]: done

**Review by riga — APPROVED** [review]
LGTM!

**Pull request merged**